src/js/components/OptionalEnviromentComponent.jsx

```diff
@@ -14,6 +14,7 @@ var OptionalEnvironmentComponent = React.createClass({
 
   propTypes: {
     errorIndices: React.PropTypes.array,
+    generalError: React.PropTypes.string,
```

Comment:
But it looks completely the same and then we need to pass responseErrorMessages too.